wire/internal/wire/parse.go

```diff
@@ -38,6 +38,37 @@ type providerSetSrc struct {
 	Import   *ProviderSet
 }
 
+// trace returns a slice of strings describing the (possibly recursive) source
+// of p, including line numbers.
+func (p *providerSetSrc) trace(fset *token.FileSet, typ types.Type) []string {
+	quoted := func(s string) string {
+		if s == "" {
+			return ""
+		}
+		return fmt.Sprintf("%q ", s)
+	}
+	var retval []string
+	if p.Provider != nil {
+		kind := "provider"
+		if p.Provider.IsStruct {
+			kind = "struct provider"
+		}
+		retval = append(retval, fmt.Sprintf("%s %s(%s)", kind, quoted(p.Provider.Name), fset.Position(p.Provider.Pos)))
+	} else if p.Binding != nil {
+		retval = append(retval, fmt.Sprintf("wire.Bind (%s)", fset.Position(p.Binding.Pos)))
```

Comment:
This could include the types involved.

Reply:
I don't think that's necessary; the line number is there so it should be easy to find, and the wire.Bind type that's being provided should be the same as the one that's the duplicate anyway.